### DISCUSSION TITLE
What is the reason for using JSON instead of TOML in zed's settings file？

### DISCUSSION THREAD

**Discussion by Curious-r** [zed-industries/zed#6481] [Feature Requests]
VSCode uses JSON, Atom uses CSON, and the TOML format tends to be relatively common throughout the Rust community for configuration. The latter two formats seem to be more friendly than JSON, so I'm curious about the reasons for this decision.

2026/08/02: 
Stumbled upon this discussion again, and it's amazing how time flies—now I'm a total TOML hater.

**Comment by petros** [comment]
This is me totally speculating here. Zed is indeed written in Rust. However, it's not a Rust only editor. The configuration file is something many folks will see and possibly edit. Folks not necessarily tied to Rust. Out of all formats, JSON is generally ubiquitous. Granted, not the most readable or easy to edit. But, nevertheless, common out there. I believe that's the reason, but I might be missing something. Happy to see someone else from Zed chiming in 😄.

**Reply by Curious-r** [reply]
If the settings file is only used to save the results of the GUI settings interface, I think it's OK. If Zed only provide the settings file for a long time, it will be a little poor experience.
Thank you for your reply! 😁Other people's views are also welcome.
